cln_lspd.go

```diff
@@ -158,7 +160,7 @@ func OnHtlcAccepted(event *glightning.HtlcAcceptedEvent) (*glightning.HtlcAccept
 	}
 
 	// fail htlc in case fetching payment information fails.
-	paymentHash, paymentSecret, destination, incomingAmountMsat, outgoingAmountMsat, fundingTxID, fundingTxOutnum, err := paymentInfo(paymentHashBytes)
+	paymentHash, paymentSecret, destination, _, outgoingAmountMsat, fundingTxID, fundingTxOutnum, err := paymentInfo(paymentHashBytes)
```

Comment:
We are calling paymentInfo twice (here and inside the lock). I guess the idea was to skip the lock if not necessary but I think the lock overhead here is negligible and the code can be easier to understand if we will only call `paymentInfo` once inside the lock.
IMO we can use the `singleflight.Group` right here (directly in the `onHtlcAccepted` function) wrapping the paymentInfo and clnOpenChannel call all together.
What do you think?

Reply:
I was thinking there's 2 scenario's. Either we open a channel or we don't.
If we don't need to open a channel, we should be superfast and parallel.
But if we do need to open a channel, it could be OK to take slightly longer.
That's why I did it this way.

Locking on the payment hash will make all MPP HTLCs go one by one. 

So in the end it's a tradeoff whether MPP is slower by (parts * algorithm duration) or channel opens slower by an additional database call.

Reply:
> Locking on the payment hash will make all MPP HTLCs go one by one.

In the case that we have a fundingTx (e.g the channel is opened) Only the query to the db will be locked for the same hash and nothing prevents from the rest of the code to run in parallel.
In theory you are right about the performance penalty but in practice I think there won't be a penalty ( we are talking about maybe several milliseconds for a db read ) and (at least IMO) the clarity and simplicity of the code wins the tradeoff here.